doc/source/tune/tutorials/tune_get_data_in_and_out.md

```diff
@@ -0,0 +1,685 @@
+
+Instead, use strings or other identifiers as your values, and initialize/load the objects inside your Trainable directly depending on those.
+
+```{note}
+[Ray Datasets](datasets_getting_started) can be used as values in the search space directly.
+```
+
+In our example, we want to tune the two model hyperparameters. We also want to set the number of epochs, so that we can easily tweak it later. For the hyperparameters, we will use the `tune.uniform` distribution. We will also modify the `training_function` to obtain those values from the `config` dictionary.
+
+
+```python
+def training_function(config):
+    # For now, we have nothing here.
+    data = None
+
+    model = {
+        "hyperparameter_a": config["hyperparameter_a"],
+        "hyperparameter_b": config["hyperparameter_b"],
+    }
+    epochs = config["epochs"]
+
+    # Simulate training & evaluation - we obtain back a "metric" and a "trained_model".
+    for epoch in range(epochs):
+        # Simulate doing something expensive.
+        time.sleep(1)
+        metric = (0.1 + model["hyperparameter_a"] * epoch / 100) ** (
+            -1
+        ) + model["hyperparameter_b"] * 0.1 * data["A"].sum()
+        trained_model = {"state": model, "epoch": epoch}
+```
+
+### Parameters
+
+```{note}
+TL;DR - use the `tune.with_parameters` util function to specify large constant parameters.
+```
+
+If we have large objects that are constant across Trials, we can use the [`tune.with_parameters`](tune-with-parameters) utility to pass them into the Trainable directly. The objects will be stored in the [Ray object store](serialization-guide) so that each Trial worker may access them to obtain a local copy to use in its process.
+
+```{warning}
+Objects put into the Ray object store must be serializable.
+```
+
+Note that the serialization (once) and deserialization (for each Trial) of large objects may incur a performance overhead.
+
+In our example, we will pass the `data` DataFrame using `tune.with_parameters`. In order to do that, we need to modify our function signature to include `data` as an argument.
+
+
+```python
+def training_function(config, data):
+    model = {
+        "hyperparameter_a": config["hyperparameter_a"],
+        "hyperparameter_b": config["hyperparameter_b"],
+    }
+    epochs = config["epochs"]
+
+    # Simulate training & evaluation - we obtain back a "metric" and a "trained_model".
+    for epoch in range(epochs):
+        # Simulate doing something expensive.
+        time.sleep(1)
+        metric = (0.1 + model["hyperparameter_a"] * epoch / 100) ** (
+            -1
+        ) + model["hyperparameter_b"] * 0.1 * data["A"].sum()
+        trained_model = {"state": model, "epoch": epoch}
+
+
+tuner = Tuner(
```

Comment:
We should move this tuner with param space to the previous section to show how to specify `param_space`.